### DISCUSSION TITLE
Set name of PDF file

### DISCUSSION THREAD

**Discussion by ericbf** [julianhille/MuhammaraJS#308] [Q&A] [answered]
Hello! We are trying to serve a MuhammaraJS PDF file from express, but change the title (by default the browser will use the last part of the URL). I don’t see an option to set the title metadata, but maybe I am missing it. Any ideas?

--------

I edited the above to say `title` instead of `filename`, as that’s what I was really referring to.

**Comment by julianhille** [comment]
The file name is set in the attachment config of express. This is not to be set on the pdf metadata level. 

If you are using `.download` from express there is a file name option. If you are streaming you need to Seth the file name in the content-disposition header.

**Reply by ericbf** [reply]
We are streaming the file using pdf writer for response.

The issue with those is that they cause the PDF to download without attempting to display in the browser. The requirement from the client is for the PDF to display in the browser, and some browsers display a filename (chrome and Firefox at least). I read online that that name is from the metadata within the file. Is there a way to set that?

**Reply by julianhille** [reply, marked as answer]
ahh ok not the file name on disk, but what the browser shows. gotcha.

https://github.com/julianhille/MuhammaraJS/blob/fb3474c2502a7925a46813ee8ad6b13d0ebcc596/lib/recipe/info.js#L93

have a look at receipt. It will help you either directly, you maybe already use recipe, or it may gives you some directions as it is implemented there.

Let me know if that helps you out.

**Reply by ericbf** [reply]
Cool, that definitely was very helpful!

For my use case, it was as simple as this:

```ts
const info = output.getDocumentContext().getInfoDictionary()

info.title = `New Title.pdf`
```

Where `output` is the writer.
